numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -468,51 +470,79 @@ def __getitem__(self, args):
             global_range and local_range attributes initialized.
 
         """
-        if isinstance(args, int):
-            self._global_range = [args]
-            self._local_range = None
-        elif isinstance(args, tuple) or isinstance(args, list):
-            if len(args) == 1 and all(isinstance(v, int) for v in args):
-                self._global_range = list(args)
-                self._local_range = None
-            elif len(args) == 2:
-                gr = args[0]
-                lr = args[1]
-                if isinstance(gr, int):
-                    self._global_range = [gr]
-                elif len(gr) != 0 and all(isinstance(v, int) for v in gr):
-                    self._global_range = list(gr)
-                else:
-                    raise IllegalRangeValueError(kernel_name=self.kernel_name)
+        warn(
+            """
+            The current syntax for specification of kernel lauch parameters
+            will be deprecated soon. Users are requested to do it through
```

Comment:
```suggestion
            is deprecated. Users should set the kernel launch arguments through
```

Reply:
Moreover, why should I get the deprecation warning if I am using the right API. The warning should be moved into the else block.

Reply:
Fixed.